modules/imgcodecs/include/opencv2/imgcodecs.hpp

```diff
@@ -227,6 +227,19 @@ Currently, the following file formats are supported:
 */
 CV_EXPORTS_W Mat imread( const String& filename, int flags = IMREAD_COLOR );
 
+
+/** @brief Loads an image from a file.
+
+This is an overloaded member function, provided for convenience. It differs from the above function only in what argument(s) it accepts and the return value.
+@param filename Name of file to be loaded.
+@param img object in which the image will be loaded.
+@param flags Flag that can take values of cv::ImreadModes
+@note
+The image passing trought the img parameter can be pre-allocated. The memory is reused if the shape and the type match with the load image.
+ */
+CV_EXPORTS_W void imread( const String& filename, Mat & img, int flags = IMREAD_COLOR );
```

Comment:
Why `OutputArray dst` doesn't work? What is the error message?

Reply:
/Users/thewoz/Documents/GitHub/opencv/modules/imgcodecs/src/loadsave.cpp:657:5: error: no matching function for call to 'imread_'
    imread_( filename, flags, dst );
    ^~~~~~~
/Users/thewoz/Documents/GitHub/opencv/modules/imgcodecs/src/loadsave.cpp:395:1: note: candidate function not viable: no known conversion from 'const _OutputArray' to 'Mat &' for 3rd argument
imread_( const String& filename, int flags, Mat& mat )
^
1 error generated.

Reply:
Use `.getMat` method of `OutputArray`